### PR TITLE
Detect deprecated methods from the interface as well.

### PR DESCRIPTION
**Pull request by eiriksm**
Based on feedback from phpstan/phpstan-deprecation-rules#49

Initial bug report on that repo: phpstan/phpstan-deprecation-rules#48

### CODE REVIEW

File: tests/PHPStan/Reflection/Annotations/DeprecatedAnnotationsTest.php [hunk 1]
```diff
@@ -131,4 +131,15 @@ public function testNonDeprecatedNativeFunctions(): void
...
 
+	public function testDeprecatedMethodsFromInterface(): void
+	{
+		require_once __DIR__ . '/data/annotations-deprecated.php';
+
+		$reflectionProvider = $this->createReflectionProvider();
+
+		$class = $reflectionProvider->getClass(\DeprecatedAnnotations\DeprecatedBar::class);
+		$scope = $this->createMock(Scope::class);
+		$this->assertTrue($class->getMethod('superDeprecated', $scope)->isDeprecated()->yes());
```
Comment (ondrejmirtes) [inline tests/PHPStan/Reflection/Annotations/DeprecatedAnnotationsTest.php:142]: I don't like mocking, you can call `getNativeMethod` instead

File: tests/PHPStan/Reflection/Annotations/DeprecatedAnnotationsTest.php [hunk 2]
```diff
@@ -131,4 +131,15 @@ public function testNonDeprecatedNativeFunctions(): void
 		$this->assertFalse($reflectionProvider->getFunction(new Name('function_exists'), null)->isDeprecated()->yes());
 	}
 
+	public function testDeprecatedMethodsFromInterface(): void
+	{
+		require_once __DIR__ . '/data/annotations-deprecated.php';
```
Comment (ondrejmirtes) [inline tests/PHPStan/Reflection/Annotations/DeprecatedAnnotationsTest.php:136]: The require doesn't need to be there. You just need to make sure to run `composer dump` locally before running tests.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

**Comment by ondrejmirtes** [comment]
Can you please rebase and force push so that the history is linear with no merge commits? Thank you.

**Comment by eiriksm** [comment]
Of course, no problem. Thanks for your time Ondrej, and thanks again for the family of phpstan libraries!

**Comment by ondrejmirtes** [comment]
Thank you!

**Comment by ossinkine** [comment]
@ondrejmirtes @eiriksm Actually this is wrong behavour and conflicts with https://github.com/phpstan/phpstan/issues/3378. The assumption that if a method in a parent class is deprecated, then a method in a child class should also be deprecated is not correct.

I have an example. [`Doctrine\ORM\Event\LifecycleEventArgs`](https://github.com/doctrine/orm/blob/152c04c03d68d39f485d367713dd69dec0f4106d/lib/Doctrine/ORM/Event/LifecycleEventArgs.php) extends [`Doctrine\Persistence\Event\LifecycleEventArgs`](https://github.com/doctrine/persistence/blob/0130920efd696cf5480744f499672b2d4ca47c85/lib/Doctrine/Persistence/Event/LifecycleEventArgs.php). Parent class has deprecated `getEntity` method and child class contain it but in child class it is not deprecated and should not be so.

So I suggest to revert this changes.

**Pull request merged**